src/pages/landing/onboarding/step2.tsx

```diff
@@ -53,21 +56,37 @@ function onboardingStep2() {
   }
 
   const handleSubmit = async () => {
-    if (account.address && id && user.id) {
+    if (account.address && builderPlaceId && user.id && userId) {
       setIsSubmitting(true);
       try {
-        const response = await setOwner({
-          id: id as string,
+        /**
+         * @dev: Update BuilderPlace first so the function will revert in case the user already owns a domain
+         */
+        const builderPlaceResponse = await setBuilderPlaceOwner({
```

Comment:
so we have two backend calls to update the BP. Could be good to have only one for perf